src/classes/USER_FrozenInActiveUser_TDTM.cls

```diff
@@ -0,0 +1,158 @@
+    public void onBeforeUpdate() {}
+
+
+	/*********************************************************************************************************
+    * @description Handles the TDTM_Runnable.Action for BeforeDelete 
+    * @return void
+    **********************************************************************************************************/
+    public void onBeforeDelete() {}
+
+
+	/*********************************************************************************************************
+    * @description Handles the TDTM_Runnable.Action for AfterInsert 
+    * @return void
+    **********************************************************************************************************/
+    public void onAfterInsert() {}
+
+
+	/*********************************************************************************************************
+    * @description Handles the TDTM_Runnable.Action for AfterUpdate 
+    * @return void
+    **********************************************************************************************************/
+    public void onAfterUpdate(List<SObject> newlist) {
+		handleIsActiveIsFrozenStatusChanges(newlist);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Handles the TDTM_Runnable.Action for AfterDelete 
+    * @return void
+    **********************************************************************************************************/
+    public void onAfterDelete() {}
+
+
+	/*********************************************************************************************************
+    * @description Handles the TDTM_Runnable.Action for AfterUndelete 
+    * @return void
+    **********************************************************************************************************/
+    public void onAfterUnDelete() {}
+
+
+	/*********************************************************************************************************
+    * @description Handles IsActive/IsFrozen changes for Users.
+    * @param newList List of Users from trigger new.
+    * @return void
+    **********************************************************************************************************/
+    private void handleIsActiveIsFrozenStatusChanges(List<SObject> newlist) {
+		Map<Id, Boolean> userIdToIsSysAdmin = USER_UserService.isSysAdmin(newlist);
+		Map<Id, Boolean> userIdToIsActive = USER_UserService.isActiveUser(newlist);
+		Map<Id, Boolean> userIdToIsFrozen = USER_UserService.isFrozenUser(newlist);
+
+		Set<Id> sysAdminUserIds = userIdToIsSysAdmin.keySet();
+		for(Id currentUserId : sysAdminUserIds) {
+			if(userIdToIsSysAdmin.get(currentUserId)      
+				&& (!userIdToIsActive.get(currentUserId)
+				 	|| userIdToIsFrozen.get(currentUserId))) {
+
+				USER_UserService.applySysAdminIsActiveIsFrozenStatusChanges(sysAdminUserIds);
```

Comment:
shouldn't you just be adding the frozen/inactive/ sysadmins to your Set, and then after your for loop, call your service with the resultant Set?  you are currently calling your service multiple times with an ever growing set.

Reply:
Updated the code accordingly. Good catch!